model/riscv_insts_vext_utils.sail

```diff
@@ -0,0 +1,1165 @@
+function nan_unbox_vector(regval, 'm) = {
+  match 'm {
+    16 => nan_unbox(regval),
+    32 => nan_unbox(regval),
+    64 => regval
+  }
+}
+
+/* Floating point classification functions */
+val f_is_neg_inf : forall 'm, 'm in {16, 32, 64}. bits('m) -> bool
+function f_is_neg_inf(xf) = {
+  match 'm {
+    16 => f_is_neg_inf_H(xf),
+    32 => f_is_neg_inf_S(xf),
+    64 => f_is_neg_inf_D(xf)
+  }
+}
+
+val f_is_neg_norm : forall 'm, 'm in {16, 32, 64}. bits('m) -> bool
+function f_is_neg_norm(xf) = {
+  match 'm {
+    16 => f_is_neg_norm_H(xf),
+    32 => f_is_neg_norm_S(xf),
+    64 => f_is_neg_norm_D(xf)
+  }
+}
+
+val f_is_neg_subnorm : forall 'm, 'm in {16, 32, 64}. bits('m) -> bool
+function f_is_neg_subnorm(xf) = {
+  match 'm {
+    16 => f_is_neg_subnorm_H(xf),
+    32 => f_is_neg_subnorm_S(xf),
+    64 => f_is_neg_subnorm_D(xf)
+  }
+}
+
+val f_is_neg_zero : forall 'm, 'm in {16, 32, 64}. bits('m) -> bool
+function f_is_neg_zero(xf) = {
+  match 'm {
+    16 => f_is_neg_zero_H(xf),
+    32 => f_is_neg_zero_S(xf),
+    64 => f_is_neg_zero_D(xf)
+  }
+}
+
+val f_is_pos_zero : forall 'm, 'm in {16, 32, 64}. bits('m) -> bool
+function f_is_pos_zero(xf) = {
+  match 'm {
+    16 => f_is_pos_zero_H(xf),
+    32 => f_is_pos_zero_S(xf),
+    64 => f_is_pos_zero_D(xf)
+  }
+}
+
+val f_is_pos_subnorm : forall 'm, 'm in {16, 32, 64}. bits('m) -> bool
+function f_is_pos_subnorm(xf) = {
+  match 'm {
+    16 => f_is_pos_subnorm_H(xf),
+    32 => f_is_pos_subnorm_S(xf),
+    64 => f_is_pos_subnorm_D(xf)
+  }
+}
+
+val f_is_pos_norm : forall 'm, 'm in {16, 32, 64}. bits('m) -> bool
+function f_is_pos_norm(xf) = {
+  match 'm {
+    16 => f_is_pos_norm_H(xf),
+    32 => f_is_pos_norm_S(xf),
+    64 => f_is_pos_norm_D(xf)
+  }
+}
+
+val f_is_pos_inf : forall 'm, 'm in {16, 32, 64}. bits('m) -> bool
+function f_is_pos_inf(xf) = {
+  match 'm {
+    16 => f_is_pos_inf_H(xf),
+    32 => f_is_pos_inf_S(xf),
+    64 => f_is_pos_inf_D(xf)
+  }
+}
+
+val f_is_SNaN : forall 'm, 'm in {16, 32, 64}. bits('m) -> bool
+function f_is_SNaN(xf) = {
+  match 'm {
+    16 => f_is_SNaN_H(xf),
+    32 => f_is_SNaN_S(xf),
+    64 => f_is_SNaN_D(xf)
+  }
+}
+
+val f_is_QNaN : forall 'm, 'm in {16, 32, 64}. bits('m) -> bool
+function f_is_QNaN(xf) = {
+  match 'm {
+    16 => f_is_QNaN_H(xf),
+    32 => f_is_QNaN_S(xf),
+    64 => f_is_QNaN_D(xf)
+  }
+}
+
+val f_is_NaN : forall 'm, 'm in {16, 32, 64}. bits('m) -> bool
+function f_is_NaN(xf) = {
+  match 'm {
+    16 => f_is_NaN_H(xf),
+    32 => f_is_NaN_S(xf),
+    64 => f_is_NaN_D(xf)
+  }
+}
+
+/* Scalar register shaping for floating point operations */
+val get_scalar_fp : forall 'n, 'n in {16, 32, 64}. (regidx, int('n)) -> bits('n) effect {escape, rreg}
+function get_scalar_fp(rs1, SEW) = {
+  if sizeof(flen) >= SEW then {
+    /* Least significant SEW bits */
+    nan_unbox_vector(F(rs1), SEW)
+  } else {
+    assert(false, "invalid vector floating-point type width: FLEN < SEW");
+    zeros()
+  }
```

Comment:
```suggestion
  assert(sizeof(flen) >= SEW, "invalid vector floating-point type width: FLEN < SEW");
  /* Least significant SEW bits */
  match SEW {
    16 => F_H(rs1),
    32 => F_S(rs1),
    64 => F_D(rs1)
  }  
```
(or whatever equivalent code compiles) matches the rest of the style and reuses the existing NaN-boxing from F_*